### ISSUE TITLE
Active development? Enhance ZstdFrameDecoder for max awesomeness

### ISSUE THREAD

**Issue by DinoCassowary** [airlift/aircompressor#137]
Currently the ZstdFrameDecoder has a lot of the functionality required for someone to build a streaming decoder (thinking to integrate into Netty channel). However, it could use a little refactoring to give visibility into the block header data and critically, https://github.com/airlift/aircompressor/blob/master/src/main/java/io/airlift/compress/zstd/ZstdFrameDecompressor.java is not a public class!

If this library is meant to be actively maintained, I would be happy to contribute, but won't if it's unlikely to get merged or be supported.

**Comment by martint** [comment]
@DinoCassowary, yes. This project is actively maintained. We'd welcome your contribution!

**Comment by DinoCassowary** [comment]
Thanks for the quick response, @martint ! I forked the repo and have started working towards this.
